src/games/supported/BankHeist.cpp

```diff
@@ -130,3 +133,38 @@ void BankHeistSettings::loadState(Deserializer & ser) {
   m_lives = ser.getInt();
 }
 
+//Returns a list of mode that the game can be played in.
+ModeVect BankHeistSettings::getAvailableModes(){
+    ModeVect modes(8);
+    for(unsigned i=0;i<8;i++){
+        modes[i] = i * 4;
+    }
+    return modes;
+}
+
+//Set the mode of the game. The given mode must be one returned by the previous function. 
+void BankHeistSettings::setMode(game_mode_t m, System &system, StellaEnvironment& environment){
+        if(m <= 28 && m % 4==0){ /*m >= 0 is implicit, since m is an unsigned int*/
+        m_mode = m;
+        // Read the mode we are currently in
+        unsigned char mode = readRam(&system,0);
```

Comment:
0 is converted to 0x80 in readRam. IIRC there are several places in which the offset is used vs. where it is not used.